File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/items/files/uri/IStoredFileUriProvider.java
package com.lasthopesoftware.bluewater.client.stored.library.items.files.uri;

import android.net.Uri;
import com.lasthopesoftware.bluewater.client.library.items.media.files.ServiceFile;
import com.namehillsoftware.handoff.promises.Promise;

public interface IStoredFileUriProvider {
	Promise<Uri> promiseStoredFileUri(ServiceFile serviceFile);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/items/files/uri/IStoredFileUriProvider.kt
package com.lasthopesoftware.bluewater.client.stored.library.items.files.uri

import android.net.Uri
import com.lasthopesoftware.bluewater.client.browsing.files.ServiceFile
import com.namehillsoftware.handoff.promises.Promise

interface IStoredFileUriProvider {
    fun promiseStoredFileUri(serviceFile: ServiceFile?): Promise<Uri?>?
}
